pyproject.toml

```diff
@@ -20,19 +20,17 @@ classifiers = [
 
 dependencies = [
     "pydantic~=2.2",
-    "optimade[server,ase]~=1.1",
+    "optimade[server]~=1.1",
     "pyyaml~=6.0",
-    "pymatgen>=2023.9",
-    "pandas >= 1.5, < 3",
-    "pybtex~=0.24",
     "tqdm~=4.65",
     "requests~=2.31",
     "numpy >= 1.22, < 3",
     "click~=8.1"
 ]
 
 [project.optional-dependencies]
-tests = ["pytest~=7.4", "pytest-cov~=4.0"]
+ingest = ["ase ~= 3.22", "pymatgen >= 2023.9", "pandas >= 1.5, < 3"]
```

Comment:
I'm not sure if these are good to separate into an optional dependency, as the main functionality (i.e. most examples, apart from the "direct jsonl" ones) will be broken if you don't install them. Thoughts?

Reply:
I guess I have a bit of a "serve-only" use case for this, where optimade-maker only fills in the provider fields etc for you and lets you programmatically launch the API more conveniently, in which case I don't want to drag in pymatgen etc. Maybe my use case is better served by just a serve wrapper around optimade-python-tools though, so if you'd prefer to keep them together then lets do that.

Reply:
ok, i see. I am now working on an aiida parser, and i think that is definitely good to separate into an optional dependency. I'm wondering, however, how to group these optional dependencies. Maybe `ase` is lightweight enough to be installed by default and the more heavy ones (`pymatgen`, `aiida`) could be optional?

Reply:
Works for me, as long as we also have an "all" extra that installs compatible versions of everything (if possible!)

Reply:
I've just separated these into extras for ase, pymatgen etc, if you think that works going forward then I think this is good to go!